modules/primo/src/Primo/Ting/Result.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * @file
+ * The Result class.
+ */
+
+namespace Primo\Ting;
+
+use Ting\Search\TingSearchRequest;
+use Ting\Search\TingSearchResultInterface;
+use TingCollection;
+
+/**
+ * A Ting compatible search result from the Primo search provider.
+ *
+ * @package Primo\Ting
+ */
+class Result implements TingSearchResultInterface {
+
+  /**
+   * The original search request that was executed to produce this result.
+   *
+   * @var \Ting\Search\TingSearchRequest
+   */
+  protected $tingSearchRequest;
+
+  /**
+   * The primo-specific search result object.
+   *
+   * @var \Primo\BriefSearch\Result
+   */
+  protected $result;
+
+  /**
+   * List of Ting collections in the search result.
+   *
+   * As Primo does not support the concept of a collection we have a 1-1
+   * relation between materials and collections.
+   *
+   * @var \TingCollection[]
+   */
+  protected $collections;
+
+  /**
+   * Result constructor.
+   *
+   * @param \Primo\BriefSearch\Result $result
+   *   Primo search result.
+   *
+   * @param \Ting\Search\TingSearchRequest $ting_search_request
+   *   Ting search query request that was executed to produce the result.
+   */
+  public function __construct($result, TingSearchRequest $ting_search_request) {
+    $this->result = $result;
+    $this->tingSearchRequest = $ting_search_request;
+
+    // Construct a Ting collection pr. document.
+    foreach ($this->result->getDocuments() as $document) {
+      $this->collections[] = new TingCollection($document->getRecordId(), new Collection([$document]));
+    }
+  }
+
+  /**
+   * Total number of elements in the search-result (regardless of limit).
+   *
+   * @return int
+   *   The number of objects.
+   */
+  public function getNumTotalObjects() {
+    return count($this->result->getDocuments());
+  }
+
+  /**
+   * Total number of collections in the search-result.
+   *
+   * Primo does not support collections so the number of collections will be
+   * equal to the number of objects.
+   *
+   * @return int
+   *   The number of collections.
+   */
+  public function getNumTotalCollections() {
+    // Will in effect be one collection pr object as we don't support
+    // collections.
+    return count($this->collections);
```

Comment:
Should this just return the result of `getNumTotalObjects()` or is this not regardless of limit?